### PR TITLE
Mixin improvements

### PR DESCRIPTION
**Pull request by mineLdiver**
Removed lots of full method overwrites that can be replaced with proper mixins and added `annoyancefix_` prefix to all mixin methods so the crashlogs (if there are any) are easier to deal with.

### CODE REVIEW

**Comment by telvarost** [comment]
Thank you for the contributions! Much appreciated.
Will these conflict with the ability to switch them on and off at runtime?
I was working on a config file for the fixes to be more compatible:
https://github.com/telvarost/AnnoyanceFix-StationAPI/issues/3
https://github.com/telvarost/AnnoyanceFix-StationAPI/compare/3-configjson-or-similar-file-for-allowing-users-to-pick-which-fixes-are-implemented-for-them?expand=1

Also, if you know how I could use gcapi to change the effective blocks list for the pickaxe and hatchet at runtime, I'm struggling to work that one out. I can only seem to change the listing on init.

File: src/main/java/net/glasslauncher/annoyancefix/mixin/HatchetMixin.java [hunk 1]
```diff
@@ -1,36 +1,49 @@
...
+    @Shadow private static BlockBase[] effectiveBlocks;
 
-    protected HatchetMixin(int i, ToolMaterial arg) {
-        super(i, 3, arg, effectiveBlocks);
+    public HatchetMixin(int i, int j, ToolMaterial arg, BlockBase[] args) {
+        super(i, j, arg, args);
+    }
+
+    @Inject(
+            method = "<clinit>",
```
Comment (telvarost) [inline src/main/java/net/glasslauncher/annoyancefix/mixin/HatchetMixin.java:24]: Does "clinit" stand for "client init" and if so, does this code only run for the client? Would it work on servers?

Comment (mineLdiver) [reply src/main/java/net/glasslauncher/annoyancefix/mixin/HatchetMixin.java:24]: `<clinit>` is the static initializer, which is everything that happens under `static {}` block, including static fields initialization. It's not sided.

**Comment by mineLdiver** [comment]
Well, these mixins will definitely conflict with current config progress, because I've basically changed everything about how the mod injects into the game. But it's still possible to add runtime checks to these.

**Comment by mineLdiver** [comment]
> I was working on a config file for the fixes to be more compatible

These new mixins are way more compatible by themselves

**Comment by 2zqa** [comment]
This PR is quite a learning experience! Thank you for taking the effort of improving this mod. I'll have to read more documentation on these ModifyConstant annotations.

**Review by telvarost — APPROVED** [review]

**Pull request merged**